apps/api/test/applicant.e2e-spec.ts

```diff
@@ -0,0 +1,197 @@
+import { IENHaPcn } from 'src/applicant/entity/ienhapcn.entity';
+import { validApplicant, applicant, addJob, seedHa, addMilestone, seedUser } from './fixture/ien';
+import { IENUsers } from 'src/applicant/entity/ienusers.entity';
+let jobTempId = 10;
+let applicantStatusId = 'NA';
+
+jest.setTimeout(10000);
+
+describe('ApplicantController (e2e)', () => {
+  let app: INestApplication;
+  let ienHaPcnRepository: Repository<IENHaPcn>;
+  let ienUsersRepository: Repository<IENUsers>;
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [AppModule],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    await app.init();
+
+    // Let's add some seed for required master table
+    ienHaPcnRepository = moduleFixture.get(getRepositoryToken(IENHaPcn));
+    await ienHaPcnRepository.upsert(seedHa, ['id']);
+
+    ienUsersRepository = moduleFixture.get(getRepositoryToken(IENUsers));
+    await ienUsersRepository.upsert(seedUser, ['id']);
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  it('Add Applicant /ien (POST)', done => {
+    request(app.getHttpServer())
+      .post('/ien')
+      .send(validApplicant)
+      .expect(res => {
+        const { body } = res;
+        applicant.id = body.id;
+      })
+      .expect(201)
+      .end(done);
+  });
+
+  it('Duplicate applicant /ien (POST)', done => {
+    request(app.getHttpServer())
+      .post('/ien')
+      .send(validApplicant)
+      .expect(res => {
+        const { body } = res;
+        expect(body.detail).toContain('already exists');
+      })
+      .expect(400)
+      .end(done);
+  });
+
+  it('Add second Applicant /ien (POST) ', done => {
+    validApplicant.applicant_id += 1;
+    validApplicant.last_name = 'notexample';
+    request(app.getHttpServer()).post('/ien').send(validApplicant).expect(201).end(done);
+  });
+
+  it('Fetch applicants list /ien (GET)', done => {
+    request(app.getHttpServer())
+      .get('/ien')
+      .expect(res => {
+        const { body } = res;
+        expect(body[1]).toBe(2);
+      })
+      .expect(200)
+      .end(done);
+  });
+
+  it('Filter applicants by name /ien?name=notexample (GET)', done => {
+    request(app.getHttpServer())
+      .get('/ien?name=notexample')
+      .expect(res => {
+        const { body } = res;
+        expect(body[1]).toBe(1);
+      })
+      .expect(200)
+      .end(done);
+  });
+
+  it('Add job/competition /ien/:id/job (POST)', done => {
+    const addJobUrl = `/ien/${applicant.id}/job`;
+    request(app.getHttpServer())
+      .post(addJobUrl)
+      .send(addJob)
+      .expect(res => {
+        const { body } = res;
+        jobTempId = body.id;
+      })
+      .expect(201)
+      .end(done);
+  });
+
+  it('Edit job/competition /ien/:id/job/:job (POST)', done => {
+    const addJobUrl = `/ien/${applicant.id}/job/${jobTempId}`;
+    addJob.job_id = 'HA100MAY11';
```

Comment:
It modifies global objects of the fixtures. It might be tricky to track the changes through the test cases.